DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -633,14 +613,22 @@ namespace pat {
 
     uint16_t packedPt_, packedEta_, packedPhi_, packedM_;
     uint16_t packedDxy_, packedDz_, packedDPhi_, packedDEta_, packedDTrkPt_;
-    uint16_t packedCovarianceDxyDxy_,packedCovarianceDxyDz_,packedCovarianceDzDz_;
-    int8_t packedCovarianceDlambdaDz_,packedCovarianceDphiDxy_;
-    int8_t packedCovarianceDptDpt_,packedCovarianceDetaDeta_,packedCovarianceDphiDphi_;
+    PackedCovariance packedCovariance_;
+
     void pack(bool unpackAfterwards=true) ;
     void unpack() const ;
     void packVtx(bool unpackAfterwards=true) ;
     void unpackVtx() const ;
+    void unpackCovarianceElement(uint16_t packed, int i,int j) const {
+      m_(i,j)= covarianceParameterization().unpack(packed,covarianceSchema_,i,j,pt(),eta(),numberOfHits(), numberOfPixelHits());
```

Comment:
OK, I see this is only called in `unpackCovariance`. I suggest moving `unpackCovarianceElement` to be `private` to avoid it being used anywhere else.